src/Microsoft.Extensions.DiagnosticAdapter/DiagnosticSourceAdapter.cs

```diff
@@ -96,7 +96,23 @@ public void Write(string diagnosticName, object parameters)
             if (!succeeded)
             {
                 var newAdapter = _methodAdapter.Adapt(subscription.MethodInfo, parameters.GetType());
-                succeeded = newAdapter(_listener.Target, parameters);
+                try
+                {
+                    succeeded = newAdapter(_listener.Target, parameters);
+                }
+                catch (InvalidOperationException ex)
```

Comment:
To be clear, the InnerException would be the new Exception and the outer exception would still be of type InvalidOperationException so as to be less breaking, right?